### PR TITLE
aliBuild init

### PR DESCRIPTION
**Pull request by dberzano**

### CODE REVIEW

**Comment by dberzano** [comment]
Closes #177.

File: aliBuild [hunk 1]
```diff
@@ -494,9 +500,57 @@ if __name__ == "__main__":
     args.noDevel = args.noDevel.split(",")
 
   # Setup build environment.
-  if not args.action == "build":
+  if args.action == "init":
+    try:
```
Comment (ktf) [inline aliBuild]: Use if.

File: aliBuild [hunk 2]
```diff
@@ -494,9 +500,57 @@ if __name__ == "__main__":
...
+        continue
+      p["ver"] = p["ver"] if p["ver"] else spec.get("tag", spec["version"])
+      debug("cloning %s%s for development" % (spec["package"], " version "+p["ver"] if p["ver"] else ""))
+      updateReferenceRepos(args.referenceSources, spec["package"], spec)
+      err = execute(format("git clone %(readRepo)s%(branch)s --reference %(refSource)s %(cd)s && " +
+                           "cd %(cd)s && git remote set-url origin %(writeRepo)s",
+                           readRepo=spec["source"],
+                           writeRepo=writeRepo,
+                           branch=" -b "+p["ver"] if p["ver"] else "",
+                           refSource=args.referenceSources+"/"+spec["package"].lower(),
```
Comment (ktf) [inline aliBuild]: use join

File: aliBuild [hunk 3]
```diff
@@ -494,9 +500,57 @@ if __name__ == "__main__":
...
+                           cd=dest))
+      dieOnError(err!=0, "cannot clone %s%s" %
+                         (spec["package"], " version "+p["ver"] if p["ver"] else ""))
+    print "Created in %s the clones for developing %s" % \
```
Comment (ktf) [inline aliBuild]: Development clones for %s created in %s

File: aliBuild [hunk 4]
```diff
@@ -422,9 +422,14 @@ if __name__ == "__main__":
                       dest="autoCleanup", action="store_false", default=True)
   parser.add_argument("--devel-prefix", "-z", nargs="?", help="Version name to use for development packages. Defaults to branch name.",
                       dest="develPrefix", default=argparse.SUPPRESS)
+
+  parser.add_argument("--dist", dest="starDist", default="",
```
Comment (ktf) [inline aliBuild]: starDist -> dist

File: aliBuild [hunk 5]
```diff
@@ -10,7 +10,7 @@ from urllib2 import urlopen, URLError
 import ssl
 import json
 
-debug, error, riemannStream = (None, None, None)
+debug, error, warning, riemannStream = (None, None, None, None)
```
Comment (ktf) [inline aliBuild]: Use a printout rather than a warning, since we want the warnings to be printed out also when we are not in debug mode.

**Comment by dberzano** [comment]
More general solution for aliBuild logging: all `print` have been replaced with `info` but the behavior stays the same (_i.e._ no `INFO:` printed for `info()` messages).

**Comment by dberzano** [comment]
All done (marked with :+1:). We should recheck, squash and merge if OK.

**Pull request merged**